core/src/test/java/org/pentaho/di/core/database/MySQLDatabaseMetaTest.java

```diff
@@ -38,16 +43,59 @@
 import org.pentaho.di.core.row.value.ValueMetaString;
 import org.pentaho.di.core.row.value.ValueMetaTimestamp;
 
+import java.sql.*;
+
 public class MySQLDatabaseMetaTest {
   MySQLDatabaseMeta nativeMeta, odbcMeta;
 
+  private DatabaseMetaData databaseMetaData;
+
+  private ResultSetMetaData resultSetMetaData, resultSetMetaDataException;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
   @Before
-  public void setupBefore() {
+  public void setupBefore() throws Exception {
     nativeMeta = new MySQLDatabaseMeta();
     nativeMeta.setAccessType( DatabaseMeta.TYPE_ACCESS_NATIVE );
+
     odbcMeta = new MySQLDatabaseMeta();
     odbcMeta.setAccessType( DatabaseMeta.TYPE_ACCESS_ODBC );
 
+    databaseMetaData = mock( DatabaseMetaData.class );
+
+    resultSetMetaData = mock( ResultSetMetaData.class );
+
+    given( resultSetMetaData.getColumnLabel( 1 ) ).willReturn( "NUMBER" );

```

Comment:
Same comments as above.

Reply:
Syntax changed from "given-willReturn" to "doReturn-when".

Reply:
Syntax changed from "when-thenThrow" to "doThrow-when".

Reply:
Created separate tests for each field.